### PR TITLE
uniform damage, multiple actions per round

### PR DESCRIPTION
**Pull request by DeinFreund**
RPS triangle works best on maps having a size > 30x30

for smaller maps range and speed have to be reduced for all units.

### CODE REVIEW

**Comment by DeinFreund** [comment]
texture filtering on android can now be enabled here: 
https://github.com/DeinFreund/SG002/blob/063580e7185f01a411b8c7b88eef6edc11325085/android/src/com/smeanox/games/sg002/android/AndroidLauncher.java#L15

seems to cause some compatibility issues though

File: core/src/com/smeanox/games/sg002/view/GameView.java [hunk 1]
```diff
@@ -166,8 +169,9 @@ private void renderGrid(SpriteBatch spriteBatch){
 	 * @param spriteBatch
 	 */
 	public void render(SpriteBatch spriteBatch, Player activePlayer){
-		aFieldSizeX = (Consts.fieldSizeX * Consts.devScaleY * zoom);
+		aFieldSizeX = (Consts.fieldSizeX * Consts.devScaleX * zoom);
```
Comment (SmBe19) [inline core/src/com/smeanox/games/sg002/view/GameView.java]: The fields should still be square and not a rectangle, so we have to use devScaleY.

File: core/src/com/smeanox/games/sg002/world/GameWorld.java [hunk 2]
```diff
@@ -148,26 +151,35 @@ public boolean isPlayerStillAlive(Player player){
...
+	 * @return
 	 */
-	public void removeGameObject(int x, int y){
-		worldMap[y][x] = null;
+	public boolean wasUsed(int y, int x){
+		try {
+			/*if (x < 0 || y < 0 || x >= mapSizeX || y >= mapSizeY) {
+				return true;//what happened here?
+			}
+			if (worldMap[x][y] == null) return true; //this shouldn't happen either*/
```
Comment (SmBe19) [inline core/src/com/smeanox/games/sg002/world/GameWorld.java]: True, this shouldn't happen. But `catch (Exception ex)` is not really nice. I think an assertion would fit better.

File: core/src/com/smeanox/games/sg002/world/GameWorld.java [hunk 3]
```diff
@@ -43,11 +43,11 @@ public int getMapSizeY() {
 		return mapSizeY;
 	}
 
-	public GameObject getWorldMap(int x, int y){
+	public GameObject getWorldMap(int y, int x){
```
Comment (SmBe19) [inline core/src/com/smeanox/games/sg002/world/GameWorld.java]: I would prefer to leave it like it was. In my opinion it makes more sense.

**Comment by DeinFreund** [comment]
I commited a bit too much there. Feel free to revert these changes.

The try/catch was for debugging only, this is how the method is supposed to look:
	public boolean wasUsed(int y, int x){
		if (x < 0 || y < 0 || x >= mapSizeX || y >= mapSizeY) {
			return true;//what happened here?
		}
		if (worldMap[x][y] == null) return true; //this shouldn't happen either*/
		return !worldMap[x][y].isCanDoAction(Action.ActionType.MOVE) &&
				!worldMap[x][y].isCanDoAction(Action.ActionType.PRODUCE) &&
				!worldMap[x][y].isCanDoAction(Action.ActionType.FIGHT);
		
	}

The unexpected cases could be logged, but shouldn't cause the game to crash.

**Comment by SmBe19** [comment]
This exception shouldn't occur as there is always a check before this method is called (as of now).

**Pull request merged**